modules/common/react/lib/utils/colorUtils.ts

```diff
@@ -8,34 +9,20 @@ export const expandHex = (hex: string) => {
   });
 };
 
-const hexToRgb = (hex: string) => {
-  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(expandHex(hex));
-  return result
-    ? {
-        r: parseInt(result[1], 16),
-        g: parseInt(result[2], 16),
-        b: parseInt(result[3], 16),
-      }
-    : null;
-};
-
 /**
  *
- * Chooses luminance color depending on the rgb value of the background color.
- * Eventually should be replaced by by Chroma 2.0
+ * Chooses foreground color with highest contrast against background
+ * (https://www.w3.org/TR/WCAG20-TECHS/G18.html)
  */
 export const pickForegroundColor = (
   background: string,
   darkColor: string = colors.blackPepper600,
   lightColor: string = colors.frenchVanilla100
 ) => {
-  const rgbColor = hexToRgb(background);
-  if (rgbColor) {
-    const r: number = rgbColor.r;
-    const g: number = rgbColor.g;
-    const b: number = rgbColor.b;
-    /** Based on : https://www.w3.org/TR/WCAG20-TECHS/G18.html */
-    return r * 0.299 + g * 0.587 + b * 0.114 > 186 ? darkColor : lightColor;
+  if (chroma.valid(background)) {
+    return chroma.contrast(background, darkColor) >= chroma.contrast(background, lightColor)
```

Comment:
Hmm - this is interesting. When they are really close sometimes it is choosing the opposite color than what we expect. For example:

![image](https://user-images.githubusercontent.com/908478/83901623-efef3200-a70f-11ea-9c1e-7f28ba5a7b84.png)

(see https://github.com/Workday/canvas-kit/blob/prerelease/v4/modules/icon/react/lib/SystemIconCircle.tsx#L72)

White is ~4.5 and black is ~4.6. Oddly enough, the contrast for white seems stronger. I think our old calculation favored white on colors vs. just choosing the highest ratio. Will have to think about how to handle this.

Reply:
Hey @donovangini - I caught up with the design team and we have a path forward. Can we update this to always select `lightColor` if its contrast is >= 4.5, otherwise take the highest of the two?

Reply:
cool, yep! updated and added a couple tests